models/buildCluster.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Command = require('../config/command');
+const scmContext = Joi.reach(require('./pipeline').base, 'scmContext');
+const scmOrganization = Joi
+    .string().max(100)
+    .description('SCM organization name')
+    .example('screwdriver-cd');
+
+const MODEL = {
+    id: Joi
+        .number().integer().positive()
+        .description('Identifier of this Job')
+        .example(123345),
+
+    name: Joi
+        .string().regex(/^[\w-]+$/)
+        .max(50)
+        .description('Name of the build cluster')
+        .example('iOS'),
+
+    description: Joi
+        .string().max(100)
+        .description('Description of the build cluster')
+        .example('Build cluster for iOS team'),
+
+    scmContext,
+
+    scmOrganizations: Joi.array().items(scmOrganization),
+
+    isActive: Joi
+        .boolean()
+        .description('Flag if the the build cluster is active')
+        .example(true)
+        .default(true),
+
+    managedBy: Joi
+        .string().valid([
+            'screwdriver',
+            'external'
+        ])
+        .description('Current state of the Job')
+        .example('screwdriver')
+        .default('screwdriver'),
+
+    managedByEmail: Command.maintainer
+};
+
+module.exports = {
+    /**
+     * All the available properties of BuildCluster
+     *
+     * @property base
+     * @type {Joi}
+     */
+    base: Joi.object(MODEL).label('BuildCluster'),
+
+    /**
+     * Properties for BuildCluster that will come back during a GET request
+     *
+     * @property get
+     * @type {Joi}
+     */
+    get: Joi.object(mutate(MODEL, [
+        'id', 'name', 'scmContext', 'scmOrganizations', 'isActive', 'managedBy', 'managedByEmail'
+    ], [
+        'description'
+    ])).label('Get BuildCluster'),
+
+    /**
+     * Properties for BuildCluster that will come back during a UPDATE request
```

Comment:
Properties for BuildCluster that will ~come back~`be passed` during a UPDATE request